src/index.js

```diff
@@ -165,13 +165,58 @@ const isHidden = function (node, displayCheck) {
   return false;
 };
 
+// form fields (nested) inside a disabled fieldset are not focusable/tabbable
+//  unless they are in the _first_ <legend> element of the top-most disabled
+//  fieldset
+const isDisabledFromFieldset = function (node) {
+  if (
+    isInput(node) ||
+    node.tagName === 'SELECT' ||
+    node.tagName === 'TEXTAREA' ||
+    node.tagName === 'BUTTON'
+  ) {
+    let parentNode = node.parentElement;
+    while (parentNode) {
+      if (parentNode.tagName === 'FIELDSET' && parentNode.disabled) {
+        // look for the first <legend> as an immediate child of the disabled
+        //  <fieldset>: if the node is in that legend, it'll be enabled even
+        //  though the fieldset is disabled; otherwise, the node is in a
+        //  secondary/subsequent legend, or somewhere else within the fieldset
+        //  (however deep nested) and it'll be disabled
+        for (let i = 0; i < parentNode.children.length; i++) {
+          const child = parentNode.children.item(i);
+          if (child.tagName === 'LEGEND') {
```

Comment:
I don't think it is necessary to loop over all elements since it's only valid to use the legend as first element of the fieldset:
https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element

Apart from that minor thing, this looks good!

Reply:
I don't like the loop, but I don't know how else to check. Note, however, that this is only looping over immediate children of the first-found disabled fieldset up the hierarchy from the node in question, and then looking for the first-found legend as an immediate (not deep) child of that fieldset to check (now that may be quite deep, but how else to check without going deep? hopefully `contains` is efficient in that regard) if the node in question is a child of it. Early breaks in all cases.

Reply:
You're right, I can't find a more performant way either!

Reply:
Thanks for checking it out!